src/librustdoc/html/render/print_item.rs

```diff
@@ -499,11 +498,7 @@ fn item_module(w: &mut Buffer, cx: &Context<'_>, item: &clean::Item, items: &[cl
                     class = myitem.type_(),
                     unsafety_flag = unsafety_flag,
                     href = item_path(myitem.type_(), myitem.name.unwrap().as_str()),
-                    title = [myitem.type_().to_string(), full_path(cx, myitem)]
-                        .iter()
-                        .filter_map(|s| if !s.is_empty() { Some(s.as_str()) } else { None })
-                        .collect::<Vec<_>>()
-                        .join(" "),
+                    title = format_args!("{} {}", myitem.type_(), full_path(cx, myitem)),
```

Comment:
looked a little deeper at this and turned out neither `myitem.type_()` or `full_path(...)` can ever be empty, so the whole joined-array thing is unneeded